distributed/scheduler.py

```diff
@@ -654,6 +672,119 @@ def validate(self):
                 pdb.set_trace()
 
 
+class TaskGroup(object):
+    """ Collection tracking all tasks within a group
+
+    Keys often have a structure like ``("x-123", 0)``
+    A group takes the first section, like ``"x-123"``
+
+    See also
+    --------
+    TaskPrefix
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.states = {state: 0 for state in ALL_TASK_STATES}
+        self.states["forgotten"] = 0
+        self.dependencies = set()
```

Comment:
> If we track dependencies, should we track dependents as well? We do so for TaskState

Maybe.  I usually try to avoid tracking state if it can be computed, unless it is expensive to compute it on the fly.  I am hoping that the number of task groups is low, and so we can call `reverse_dict` when necessary to get dependents.

> What's the intention of tracking this information in the group in the first place? I do not see it being used and would suggest to only add this once it is actually used.
I can see the usefulness e.g. building high level dashboards with the dask groups as a replacement for the /graph page which becomes unusable for too large graphs anyhow.

That is my intention (although I'm also looking at replacing the task stream plot, which becomes a non-informative blur eventually).  

> I do not see it being used and would suggest to only add this once it is actually used.

In general I agree with this sentiment, but for now I would like to ignore it.  I think that it will be useful to have this around for debugging, and while I experiment with additional dashboarding, which will happen over the next month or two.

Reply:
> I usually try to avoid tracking state if it can be computed, unless it is expensive to compute it on the fly

I understand. That also makes sense for the TaskState since there is is much more difficult

> but for now I would like to ignore it.

Fine by me. I already suspected this to be preparing follow up work